### ISSUE TITLE
Cannot create kcov-solib.pipe on filesystem that does not support mkfifo

### ISSUE THREAD

**Issue by kennytm** [SimonKagstrom/kcov#139]
Say we run

```bash
$ kcov out_dir ./executable
```

If `out_dir` is mounted on a filesystem that does not support mkfifo (e.g. NFS server, Virtual box guest shared folder), the collection will fail with message

```plain
kcov-solib: Can't open out_dir/executable/kcov-solib.pipe
```

and the report will give "NaN%" coverage. But it should still be possible to collect coverage into this `out_dir` if the pipe is placed elsewhere e.g. `/tmp` right?

I think [this line](https://github.com/SimonKagstrom/kcov/blob/938f545476cd81206001cd4cc1329311ecea7df6/src/solib-handler.cc#L100) should fallback to `mkfifo` on a `mkdtemp` if it failed.

* kcov version: master (ac74935)
* target OS: Debian 8.4 amd64 on Virtual Box 5.0.20.

**Comment by SimonKagstrom** [comment]
Thanks for your report!

Yes, this would be better. It this the cause of the coverage collection failure through? I think it should be possible to collect anyway, just without shared libraries. Does it behave the same way if passing --skip-solibs?

**Comment by kennytm** [comment]
@SimonKagstrom Thanks for the suggestion! `--skip-solibs` does get rid of the error, unfortunately it also collects nothing and the coverage is 0%.

I have checked the solib.pipe is indeed the problem since I can get a valid coverage report when the output directory is set to local drive.

I am using kcov with a Rust project, following the instruction from https://users.rust-lang.org/t/tutorial-how-to-collect-test-coverages-for-rust-project/650

**Comment by SimonKagstrom** [comment]
Fixed with 15a8be8.

(I hope!)

**Comment by kennytm** [comment]
@SimonKagstrom Hi, thanks for the fix :) Unfortunately 15a8be8 does not work, because the lines

```c++
		snprintf(buf, sizeof(buf), "kcov-solibXXXXXX");
		char *dir = mkdtemp(buf); 
```

simply creates the temporary directory in the current folder, so it does not help if the whole filesystem doesn't support mkfifo. Instead, it should use an absolute path like

```c++
		const char* tmpdir = getenv("TMPDIR");
		if (!tmpdir) {
		    tmpdir = "/tmp";
		}
		snprintf(buf, sizeof(buf), "%s/kcov-solibXXXXXX", tmpdir);
		char *dir = mkdtemp(buf); 
```

**Comment by SimonKagstrom** [comment]
Sounds like there is a pull-request brewing somewhere! :-)

**Comment by SimonKagstrom** [comment]
Fixed with fff037c according to your patch!
